File: PrimaMateria/PrimaMateria/Classes/DynoTableCell.m
//
//  DynoTableCell.m
//  PrimaMateria
//
//  Created by Jerry Porter on 1/5/2011.
//  Copyright 2016 xTrensa. All rights reserved.
//

//typedef NS_ENUM(NSInteger, LabelTypes) {
//    kAtomicNumber = 1,
//    kSymbol = 2
//};

@interface DynoTableCell() {
    NSMutableArray *columns;
}

@end

@implementation DynoTableCell

- (instancetype)initWithStyle: (UITableViewCellStyle)style reuseIdentifier: (NSString *)reuseIdentifier {
    if (self = [super initWithStyle:(UITableViewCellStyle) style reuseIdentifier:reuseIdentifier]) {
        columns = [NSMutableArray arrayWithCapacity: 5];
    }
    return self;
}

- (void)addColumn: (CGFloat)position {
    [columns addObject:@(position)];
}

- (void)drawRect: (CGRect)rect {
    [super drawRect: rect];
    
    self.backgroundColor = UIColor.blackColor;
    self.contentView.backgroundColor = UIColor.darkGrayColor;
    
    if(columns.count > 0) {
        CGContextRef ctx = UIGraphicsGetCurrentContext();
        
        // just match the color and size of the horizontal line
        CGContextSetRGBStrokeColor(ctx, 0.5, 0.5, 0.5, 1.0);
        CGContextSetLineWidth(ctx, 0.25);
        
        for (NSUInteger i = 0; i <columns.count; i++) {
            // get the position for the vertical line
            CGFloat f = ( (NSNumber *)columns[i] ).floatValue;
            CGContextMoveToPoint(ctx, f, 0);
            CGContextAddLineToPoint(ctx, f, self.bounds.size.height);
        }
        CGContextStrokePath(ctx);
    }
}

#pragma mark - Memory Management Methods

@end

File: PrimaMateria/PrimaMateria/Classes/DynoTableCell.swift
//
//  DynoTableCell.swift
//  PrimaMateria
//
//  Created by Jerry Porter on 4/14/16.
//  Copyright © 2016 xTrensa. All rights reserved.
//

@objc class DynoTableCell : UITableViewCell {
    private var columns : NSMutableArray
    
    // MARK: - Initialization Methods
    
    required init?(coder aDecoder: NSCoder) {
        self.columns = NSMutableArray.init(capacity: 5)
        super.init(coder:aDecoder)!
    }
    
    override func awakeFromNib() {
        super.awakeFromNib()
        self.setup()
    }
    
    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        self.columns = NSMutableArray.init(capacity: 5)
        super.init(style: style, reuseIdentifier: reuseIdentifier)
    }
    
    // MARK: - Internal Methods
    
    func addColumn(position: CGFloat) {
        columns.addObject(position)
    }
    
    func setup() {
        self.backgroundColor = UIColor.blackColor()
        self.contentView.backgroundColor = UIColor.blackColor()
        
        if(columns.count > 0) {
            let ctx : CGContextRef = UIGraphicsGetCurrentContext()!
            
            CGContextSetRGBStrokeColor(ctx, 0.5, 0.5, 0.5, 1.0)
            CGContextSetLineWidth(ctx, 0.25)
            
            for index in 0...columns.count {
                let f : CGFloat = CGFloat(columns[index] as! NSNumber)
                CGContextMoveToPoint(ctx, f, 0)
                CGContextAddLineToPoint(ctx, f, self.bounds.size.height)
            }
            CGContextStrokePath(ctx)
        }
    }
}
